product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def _get_price_from_supplierinfo(self, partner_id):
+        if partner_id:
```

Comment:
i think you can `assert` this in order to keep it flatter

Reply:
What do you think about change the logic here:

```
        if not partner_id:
             return 0
        supplierinfo = self.env['product.supplierinfo'].search(
           ['|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
            ('product_id', '=', product.id),
            ('type', '=', 'customer'),
            ('name', '=', partner_id)])
        return supplierinfo.price
```

Whit this you gain a indentation

And maybe add a `self.onsere_one()` in the method, you has the return in the `for`